### PR TITLE
Plane: fixed stick mixing and throttle nudge in QRTL approach

### PR DESCRIPTION
**Pull request by tridge**
thanks to Henry for noticing this

### CODE REVIEW

**Comment by peterbarker** [comment]
I can understand the shape of this as useful for backporting.

But any time I see a list of modes like at the top of that `void Plane::stabilize_stick_mixing_fbw` I'm thinking a callback is a good idea.  The qrtl-specific check you've got in there to check the phase would be part of qrtl's implementation of that callback.

So `if (!control_mode->do_stabilize_stick_mixing_fbw()) {` sort of thing.

File: ArduPlane/Attitude.cpp [hunk 1]
```diff
@@ -502,6 +501,11 @@ void Plane::stabilize()
     } else if (control_mode->is_vtol_mode() && !quadplane.tailsitter.in_vtol_transition(now)) {
         // run controlers specific to this mode
         plane.control_mode->run();
+        if (control_mode == &mode_qrtl &&
+            quadplane.poscontrol.get_state() <= QuadPlane::QPOS_AIRBRAKE) {
```
Comment (IamPete1) [inline ArduPlane/Attitude.cpp:505]:
```suggestion
            (quadplane.poscontrol.get_state() == QuadPlane::QPOS_APPROACH ||
            quadplane.poscontrol.get_state() == QuadPlane::QPOS_AIRBRAKE)) {
```
We have been caught out by less than in the past and found there are edge cases where we come through this in `QPOS_NONE`

Comment (IamPete1) [reply ArduPlane/Attitude.cpp:505]: You could also put this in the run method for QRTL, its called on the line above.

https://github.com/ArduPilot/ardupilot/blob/442e7832146508718b04b5045157b672e4a95780/ArduPlane/mode_qrtl.cpp#L38-L52

Comment (tridge) [reply ArduPlane/Attitude.cpp]: thanks, fixed

**Comment by Hwurzburg** [comment]
fixes the issue I saw...thanks

**Review by Hwurzburg — APPROVED** [review]

**Review by IamPete1 — APPROVED** [review]

**Pull request merged**